File: LeetCode/2576. Find the Maximum Number of Marked Indices/solution.cpp
class Solution {
public:
    int maxNumOfMarkedIndices(vector<int>& nums) {
        sort(nums.begin(), nums.end(), greater{});
        //for (int t : nums) cout << t << " "; cout << endl;
        int ret = 0;
        for (int i = 0, n = nums.size(), j = (n + 1) / 2; j < n; ++j) {
            if (nums[i] >= nums[j] * 2) {
                ++i;
                ret += 2;
            }
        }
        return ret;
    }
};

File: LeetCode/2576. Find the Maximum Number of Marked Indices/solution.py
class Solution:
    def maxNumOfMarkedIndices(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        lo, hi = 0, n//2
        while lo <= hi:
            m = (lo+hi)//2
            if all(a*2 <= b for a, b in zip(nums[:m],nums[-m:])):
                lo = m + 1
            else:
                hi = m - 1
        return hi*2 
